independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTree.java

```diff
@@ -80,6 +80,13 @@ static PathTree ofArchive(Path archive, PathFilter filter) {
         return ArchivePathTree.forPath(archive, filter);
     }
 
+    /**
+     * Whether the PathTree provides local resources.
+     * <p>
+     * For instance a directory or a file provides local resources. A jar does not.
+     */
```

Comment:
I don't really understand what local resources are

Reply:
Local resources are resources that are local to your deployment. A jar is an external dependency so it's not local.

This is a preexisting notion as the `QuarkusClassLoader` provides the local resources to the tests: see `QuarkusClassLoader#getLocalClassNames()` in the existing code.

Reply:
Still sounds confusing to me, but if we are already using that terminology, then 🤷🏽

Reply:
I don't mind changing the name to something that makes more sense to you - I really reused the terminology that was there.
Ideally, let's do that once this patch is in.

Reply:
It should be something like `isPathsVisibleOutside()` instead to indicate whether the paths visited while processing this tree are visible outside the context of the `visit` method.

Reply:
Well, that's not the semantic I want here. Because we could make the jar content visible from the outside and that still wouldn't make the jar resources candidate for this. What I want to really know if we consider these classes local to the current deployment.

BTW, I don't mind people bike shedding on the name but please let's do it in a second step. I really want this in.

Reply:
> Well, that's not the semantic I want here. Because we could make the jar content visible from the outside and that still wouldn't make the jar resources candidate for this. What I want to really know if we consider these classes local to the current deployment.

From the PathTree perspective that's what this is though. Either the paths are visible or not.

> BTW, I don't mind people bike shedding on the name but please let's do it in a second step. I really want this in.

Sure. That's a public API though, so if it's a follow up it has to go right after this one and if this one gets rolled back, they have to be rolled back together.

Reply:
@aloubyansky I don't think you should place yourself from the PathTree perspective when you define the API: it's really for the consumers that the name is important.

For this very patch, I don't want to know if I'm able to get the content using something else than visit. I want to know if this thing is an archive and thus not something that will be subject to change.
The fact that I have to use `visit` is really an implementation detail (and could change over time) and moreover, that's not what I'm trying to do here: I'm just trying to know if the classes will be subject to change or not.

I'm not saying that my name is ideal but at least from my perspective `isPathsVisibleOutside` has an obscure semantic and I have no idea what I should use it for. And I probably won't have used it if it existed as for me it's not the same semantic.

If you come with a better name, I'll be happy to follow up after this PR is in.

Reply:
> thus not something that will be subject to change.

So why not come up with a name that reflects that?

Reply:
I'm all ears but really it can be done in a follow-up as this patch have been around for quite a while now and I want it in.
If you all can come up with a good proposal, I will be happy to do the leg work in another PR.

Reply:
> isPathsVisibleOutside has an obscure semantic

Local is a term from a higher (classloading) level. It doesn't apply well to `PathTree` abstraction. `isPathsValidOutside()` is exactly what this is.

Reply:
@gsmet is your concern about `OpenContainerPathTree`? Would `isArchiveOrigin()` be better for you?

Reply:
Well, I don't think it is, especially because of the semantic in `MultiRootPathTree` but I pushed the change as I'm a bit tired of arguing.
My personal opinion is that the semantic I chose was good and maybe the name needed some love. Maybe `providesApplicationClasses` to be on par with `QuarkusClassLoader#isApplicationClass()` as it's really the same semantic.
But I don't think the name was that bad once you understand the semantic.

I spent a lot of time on this patch and I would really appreciate if we could take a decision and merge it.

Thanks.

Reply:
Note that if you don't understand the semantic, we might better have a quick call so that I explain it to you.